### PR TITLE
JBIDE-20988: Adding test for node64.exe preference validation. Adding missing NON-NLS

### PR DESCRIPTION
**Pull request by ibuziuk**

### CODE REVIEW

**Comment by alexeykazakov** [comment]
testNode64ExePreferenceValidation(org.jboss.tools.jst.js.node.test.NodeUITestCase)  Time elapsed: 15.315 sec  <<< FAILURE!
junit.framework.AssertionFailedError: Valid node home folder - 'OK' must be enabled
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.Assert.assertTrue(Assert.java:22)
	at junit.framework.TestCase.assertTrue(TestCase.java:192)
	at org.jboss.tools.jst.js.node.test.NodeUITestCase.testNode64ExePreferenceValidation(NodeUITestCase.java:59)

**Comment by ibuziuk** [comment]
hmmm... was fine on my machine, will take a look

**Comment by ibuziuk** [comment]
was closed accidentally ;-)

**Comment by alexeykazakov** [comment]
I'm using Ubuntu if it matters.

**Comment by ibuziuk** [comment]
yeah, I'm thinking about rebooting now

**Comment by ibuziuk** [comment]
@alexeykazakov could you give it a go one more time?

**Comment by alexeykazakov** [comment]
testNode64ExePreferenceValidation(org.jboss.tools.jst.js.node.test.NodeUITestCase)  Time elapsed: 0.004 sec  <<< ERROR!
org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException: The widget was null.
	at org.eclipse.swtbot.swt.finder.widgets.AbstractSWTBot.<init>(AbstractSWTBot.java:104)
	at org.eclipse.swtbot.swt.finder.widgets.AbstractSWTBotControl.<init>(AbstractSWTBotControl.java:47)
	at org.eclipse.swtbot.swt.finder.widgets.SWTBotShell.<init>(SWTBotShell.java:49)
	at org.eclipse.swtbot.swt.finder.widgets.SWTBotShell.<init>(SWTBotShell.java:38)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.activeShell(SWTBotFactory.java:376)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.menu(SWTBotFactory.java:210)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.menu(SWTBotFactory.java:200)
	at org.jboss.tools.jst.js.node.test.NodeUITestCase.testNode64ExePreferenceValidation(NodeUITestCase.java:48)

**Comment by ibuziuk** [comment]
weird error, I saw some guys did bot.sleep(~1 sec) in the beginning of the test but it looks like irrelevant

**Comment by ibuziuk** [comment]
@alexeykazakov I do hope it will work from the 3rd go

**Comment by alexeykazakov** [comment]
testNode64ExePreferenceValidation(org.jboss.tools.jst.js.node.test.NodeUITestCase)  Time elapsed: 10.959 sec  <<< ERROR!
org.eclipse.swtbot.swt.finder.widgets.TimeoutException: Timeout after: 5000 ms.: The widget (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH') was not enabled.
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:437)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:411)
	at org.jboss.tools.jst.js.node.test.NodeUITestCase.testNode64ExePreferenceValidation(NodeUITestCase.java:65)

Sorry.

**Comment by ibuziuk** [comment]
@alexeykazakov  LOL, I know why is it failing on Linux - node64.exe is not valid executable for preferences on Linux, for Linux either *nodejs* or *node* are valid for preferences. Will update the test to make it platform independent ;-)

**Comment by ibuziuk** [comment]
build works on my Ubuntu

File: tests/org.jboss.tools.jst.js.node.test/src/org/jboss/tools/jst/js/node/test/NodeUITestCase.java [hunk 1]
```diff
@@ -40,4 +45,51 @@ public static SWTBotView getProjectExplorer() {
...
 	
+	// JBIDE-20989 Preference validation fails on windows if node executable called node64.exe
+	public void testNode64ExePreferenceValidation() {
+		getProjectExplorer().show();
+		bot.menu("Window").menu("Preferences").click(); //$NON-NLS-1$ //$NON-NLS-2$
+		bot.waitUntil(Conditions.shellIsActive("Preferences"), NodeAllTests.DEFAULT_TIMEOUT); //$NON-NLS-1$
+		bot.tree().expandNode("JavaScript Tools").select("Node"); //$NON-NLS-1$ //$NON-NLS-2$
+		
+		IFolder nodeInstallationFolder = testProject.getFolder(getNodeInstallationFolder());
+		assertNotNull("Can't load JBIDE-20989 folder", nodeInstallationFolder); //$NON-NLS-1$
```
Comment (alexeykazakov) [inline tests/org.jboss.tools.jst.js.node.test/src/org/jboss/tools/jst/js/node/test/NodeUITestCase.java]: getFolder() never returns null. So you can remove this assertNotNull. It's enough to check if nodeInstallationFolder.exists()

**Comment by alexeykazakov** [comment]
Tests passed.

**Pull request merged**